### DISCUSSION TITLE
Increasing credit for contributions

### DISCUSSION THREAD

**Discussion by paulf81** [NatLabRockies/floris#925] [General]
This discussion channel is meant to list ideas for expanding the recognition of credit to contributors to FLORIS.  This came up during the WETO software workshop, with some initial suggestions including:

- Pull requests to augment credit in code documentation
- Follow OpenOA in using all-contributors (suggestion from @RHammond2 )

Other ideas?  Flagging @misi9170 @rafmudaf @Bartdoekemeijer @bayc 

(Note whatever is adopted will be likewise brought to FLASC)

**Comment by Bartdoekemeijer** [comment]
One thought I had was the "number of lines of code changed" or "number of pull requests" leaderboard. I don't want to cause unnecessary competition, but I wanted there to be an incentive for doing more than just one commit/change. Not sure what makes sense, but a leaderboard like that could be a nice motivator for people to try to merge in bigger pull requests.

**Reply by rafmudaf** [reply]
I suppose this should filter out who is the "core" development team. Since it's our (Misha, Paul, me, a few others) job, we'll always be at the top, but the point is to highlight who has contributed back even when they don't have to.

**Reply by RHammond2** [reply]
I was just perusing this and I'm still a top contributor despite having limited development on the project, which I think is where the all-contributors pairs nicely because it can highlight the different ways people contribute to a project. That said, all-contributors is what our PI wanted  to use given the number of non-code contributions people make.

**Comment by misi9170** [comment]
I guess it depends how public we want the leaderboard to be. Github already [tracks contributions](https://github.com/NREL/floris/graphs/contributors), which does go by some sort of combination of number of commits and number of lines changed.

EDIT: possibly just by number of commits, which isn't necessarily a great metric

**Reply by paulf81** [reply]
But then there are some tools to build some nice displays maybe: https://github.com/eugef/github-leaderboard
